src/node.cpp

```diff
@@ -1863,6 +1901,22 @@ LocalNode* LocalNode::unserialize(Sync* sync, string* d)
 
     const char* localname = ptr;
     ptr += localnamelen;
+
+    bool syncable = true;
+    if (hasExtensionBytes)
+    {
+        syncable = MemAccess::get<bool>(ptr);
+        ptr += sizeof(syncable);
+
+        for (int i = 7; i--;)
+        {
+            if (ptr + (unsigned char)*ptr < end)
+            {
+                ptr += (unsigned char)*ptr + 1;
+            }
+        }
```

Comment:
this function would benefit a lot from using CacheableReader/CacheableWriter too